src/template_finder.py

```diff
@@ -63,9 +70,97 @@ def __new__(cls, save_last_res=False):
                     ]
         return cls._instance
 
+    @staticmethod
+    def _make_template(template, match, scale, roi, normalize_monitor) -> TemplateMatch:
+        """Helper function to make a TemplateMatch object from ."""
+        rx, ry, rw, rh = roi
+        template_match = TemplateMatch()
+        rec = [
+            int((match[1] + rx) // scale),
+            int((match[0] + ry) // scale),
+            int(template.shape[1] // scale),
+            int(template.shape[0] // scale)
+        ]
+        ref_point = roi_center(rec)
+
+        if normalize_monitor:
+            rec[0], rec[1] = convert_screen_to_monitor((rec[0], rec[1]))
+        else:
+            template_match.center = ref_point
+            template_match.region = rec
+            template_match.valid = True
+        return template_match
+
     def get_template(self, key):
         return cv2.cvtColor(self._templates[key][0], cv2.COLOR_BGRA2BGR)
 
+    def search_v2(
```

Comment:
Since this is my first contribution I didn't want to touch such an integral part of the program therefore I just made a `search_v2` method.
I'm not particularly excited by this approach but it will ensure the parts I touched don't interfere with the rest.
I think this logic is a bit easier to follow though.